internal/release/release.go

```diff
@@ -0,0 +1,65 @@
+package release
```

Comment:
I was on the fence about this one, I know they're doing it that way to facilitate versioning, especially multi-version support in parallel, which is very important for a Kubernetes controller, but really not so much for a CLI like us.  We will probably handle our versioning/backward compatibility more simply.  The reason I like and used this approach here is that it reduces verbosity, as the package name qualifies everything it contains (eg: `environment`, `release`).  

Just have a look at the names of all the entities defined in this file for example:
https://github.com/nestoca/joy/blob/9195dcade8750872f2602b2e1add3a9738c7d6ac/cmd/joy/environment.go
If we were to put all CRDs into the same package, we would basically need to prefix/suffix everything in it with `Environment` to prevent clashes with the elements associated with different CRDs.

Beyond "it's the way they usually do it in controllers" and the possibility to handle multiple versions in parallel (which I'm not sure we would want to get into anyway, because it has a great cost on complexity) do you really see any other major benefits?

Reply:
> The reason I like and used this approach here is that it reduces verbosity, as the package name qualifies everything it contains (eg: environment, release).

I would agree, if we didn't have to eventually manage different versions of the schema. Packing by API version will make any code that needs to process/support multiple schema versions much more readable whereas the structure now will make that process more complex IMO. 

> We will probably handle our versioning/backward compatibility more simply.
Can you elaborate on this? 

> If we were to put all CRDs into the same package, we would basically need to prefix/suffix everything in it with Environment to prevent clashes with the elements associated with different CRDs.
I don't really see an issue with this, anyone with experience writing tooling related to Kubernetes will already be used to this. 

> I'm not sure we would want to get [the possibility to handle multiple versions in parallel] into anyway
IMO thats not a decision we should take at this stage. Having the ability to handle multiple versions will allow us to introduce new features and test them with real projects / roll them out progressively. With a tool that will become critical in our infrastructure I would argue that this is non-negotiable so we don't box ourselves into a corner 

Consider also tooling that will need to depend on these schemas (Ex the generator), we don't need to make all the contents of the `release` package here available as a dependency, potentially we'll just the schema, so it will eventually be separated from this package anyway.

Reply:
OK, great! It actually simplified and solved some of the packaging issues I was having.  Still had to do some shuffling to prevent cyclic dependencies, but I'm quite satisfied with the result! 😅

See this commit:
https://github.com/nestoca/joy/pull/9/commits/86473eba7b0d16f3d42758c0761ba7f4936f1ee8